event/multisub.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package event
+
+// JoinSubscriptions joins multiple subscriptions to be able to track them as
+// one entity and collectively cancel them of consume any errors from them.
+func JoinSubscriptions(subs ...Subscription) Subscription {
```

Comment:
This is nice and neat, but you _could_ have used a much simpler one, right? You know how many subpools you have,  so you could have had one goroutines listening to all of them, instead of one routine each: 

```golang
			select {
				case err := <-sub1.Err():
					if err != nil {
						errc <- err
					}
				case err := <-sub2.Err():
					if err != nil {
						errc <- err
					}
...
```
Doesn't matter though, your solution is more generic

Reply:
The subpools can be added dynamically to the new txpool. Currently we have legacypool (and will have blobpool). We'll IMO add localpool, Marius wants to experiment with some disk backed pool. I can imagine enging up with a few controllable via a CLI flag (at least while testing implementations). It's simpler not to assume how many there will be and how they would be assembled.